Minor fixes in Semgrep Pro Engine docs, update screenshots in Findings page, update Demo Project, remove demo project from Semgrep Pro Engine docs (#880)
returntocorp/semgrep-docs: docs/deepsemgrep/deepsemgrep-introduction.md

1. Sign in to [Semgrep Cloud Platform](https://semgrep.dev/login).
1. Select **[Settings](https://semgrep.dev/orgs/-/settings)**.
1. Enable the <i class="fa-solid fa-toggle-large-on"></i> **Semgrep Pro Engine beta** toggle.
1. Ensure that you have the **default ruleset** enabled in your **[Rule Board](https://semgrep.dev/orgs/-/board)**. In the Rule Board, the **default ruleset** is in the **Monitor column**. If this ruleset is **not** added, go to [https://semgrep.dev/p/default](https://semgrep.dev/p/default), and then click **Add to Rule Board**.

> 1. Optional: If you don't have any projects added in your organization, follow the procedures described in [Scanning a repository from GitHub or GitLab](/semgrep-code/getting-started-with-semgrep-code/#semgrep-code-with-semgrep-cloud-platform) to scan a new project with Semgrep Pro Engine.

Reviewer: ```suggestion
1. Optional: If you don't have any projects added in your organization, follow the procedures described in [Scanning a repository from GitHub or GitLab](/semgrep-code/getting-started-with-semgrep-code/#semgrep-code-with-semgrep-cloud-platform) to scan a new project with Semgrep Pro Engine. Ensure that your project's language is supported by Semgrep Pro Engine.
```
Note that the projects they onboard have to be of any language supported by Semgrep Pro Engine. The reason why I used juice shop was because it's used in the CLI for Semgrep Pro.

Author: ```suggestion
1. Optional: If you don't have any projects added to your organization, follow the procedures described in [Scanning a repository from GitHub or GitLab](/semgrep-code/getting-started-with-semgrep-code/#semgrep-code-with-semgrep-cloud-platform) to scan a new project with Semgrep Pro Engine. Ensure that your project's language is supported by Semgrep Pro Engine.

:::info Testing Semgrep Pro Engine
To test Semgrep Pro Engine on a purposefully vulnerable repository, fork [juice-shop](https://github.com/juice-shop/juice-shop) repository, and then add it to SCP by following the steps described in  [Adding a repository](/semgrep-app/getting-started-with-semgrep-app/#option-b-adding-a-repository-from-github-or-gitlab).
:::
```

## Additional information

### Experimental support for interprocedural analysis

> Interprocedural analysis keeps the fast scan times of Semgrep OSS while also finding new vulnerabilities that cross functions. It's designed to improve results of pull or merge requests (PRs or MRs) and help security engineers build trust with developers.

Reviewer: I would revert the change. "shown in" is correct. The result of the semgrep scan is displayed to the users. NOT the result of a pull or merge request.

Author: ```suggestion
Interprocedural analysis keeps the fast scan times of Semgrep OSS while also finding new vulnerabilities that cross functions. It's designed to improve results shown in pull or merge requests (PRs or MRs) and help security engineers build trust with developers.
```

### Semgrep Pro Engine CI scan issues

> To provide reliability, Semgrep Pro Engine can fall back to the use of Semgrep OSS Engine. This ensures that in the vast majority of cases, scans run successfully.

Reviewer: The important bit here is that the scan is **completed**. That is more concrete than "reliable" which could mean anything.

Author: ```suggestion
To provide realibly completed scans, Semgrep Pro Engine can fall back to the use of Semgrep OSS Engine. This ensures that in the vast majority of cases, scans run successfully.
```
